### DISCUSSION TITLE
HYPNOTOAD_REV ?

### DISCUSSION THREAD

**Discussion by Timtroll** [mojolicious/mojo#1893] [Q&A]
Can someone explain why the application is re-launched when Hypnotoad is launched?
Judging by the code, the variable $ ENV {HYPNOTOAD_REV} is being reinstalled.
How safe will it be to run the application with the preset parameter $ ENV {HYPNOTOAD_REV} = 1; ?

like this:
`   $ENV { HYPNOTOAD_REV } = 1;`
```       my $ oHypnotoad = Mojo :: Server :: Hypnotoad-> new;
       $ oHypnotoad-> configure (
          'hypnotoad' => {
                   'listen' => ['http: // *: 4444'],
                   'workers' => 10
          }
       );
       $ oHypnotoad-> run ('packages / ApiWeb.pm'); ```

**Comment by kraih** [comment]
The restart happens to make sure the application works with the hot deployment feature. It's a test. Do not use undocumented environment variables.

**Reply by Timtroll** [reply]
Can you please tell me how can I run hypnotaod once and without restarting the mojo startup script?

**Reply by daleif** [reply]
I'm curious, what exactly is the problem with the restart as it is?

**Reply by kraih** [reply]
Restarting is the main feature of Hypnotoad, if you don't need it just use `prefork` instead.
